### PR TITLE
Improves expand err

### PR DESCRIPTION
**Pull request by jcchavezs**
While working with the Expand provider I struggled to understand what was my issue and even more the returning error was more confusing. Wrapping it with a meaningful context makes it much easier and allows implementors to do better assertions.

Ping @bogdandrutu

### CODE REVIEW

**Review by carlosalberto — APPROVED** [review]

File: config/configmapprovider/mock.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	retrieveErr error
+}
+
+var _ Provider = &mockProvider{}
+
+func (m *mockProvider) Retrieve(ctx context.Context, onChange func(*ChangeEvent)) (Retrieved, error) {
+	return m.retrieved, m.retrieveErr
+}
+
+func (mockProvider) Shutdown(ctx context.Context) error { return nil }
```
Comment (codeboten) [inline config/configmapprovider/mock.go:35]: I'm surprised this code isn't exercised in the tests, is a call to Shutdown missing somewhere?

File: config/configmapprovider/mock.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	getErr error
+}
+
+var _ Retrieved = &mockRetrieved{}
+
+func (sr *mockRetrieved) Get(ctx context.Context) (*config.Map, error) {
+	return sr.got, sr.getErr
+}
+
+func (mockRetrieved) Close(ctx context.Context) error { return nil }
```
Comment (codeboten) [inline config/configmapprovider/mock.go:48]: Same question as for shutdown.

File: config/configmapprovider/mock.go (hunk 1 above)
Comment (jcchavezs) [reply config/configmapprovider/mock.go:35]: Not 100% sure what you mean here. Are you asking for a test for the mock implementation?

Comment (codeboten) [reply config/configmapprovider/mock.go:35]: I'm not, I was just curious if the code that's using the mockProvider forgets to call shutdown somewhere, I would have expected it to be called in a `defer` call somewhere.

**Review by codeboten — APPROVED** [review]

**Review by bogdandrutu — APPROVED** [review]
In the future please send unrelated cleanup PRs like the package import in a different PR. Thank you

**Pull request merged**